Plugin front views and API

fvalladaresj opened the pull request:
# Summary

<!--
Include a short summary of the changes made to the platforn and list any dependencies
of the pr - Required.
-->
Implement front end for plugins system.

## Type of change

<!-- Indicate the type of change. Delete those that do not apply  - Required -->

- Front end new feature.

## Changes

<!--
Indicate the changes/fixes that you want to merge - Required.

- Bug 1
- Bug 2
- Feature 1
- Feature 2
-->
- Add plugins page
- Add plugins components
- Add pluginStatus enum constant to register the state of the plugin
- Add custom hooks for api connection
- Add sample data for json-server testing
- Add react-markdown library 

## How to Test

<!--
Describe the tests or executions that you ran to verify your changes and provide
instructions to reproduce them - Required.
-->
...

## Screenshots

<!--
In the case of modifying a view in the front-end, include screenshots with the changes.
Optional, delete this section if not needed.-->
...

## Notes

<!--
Include any additional information that would be useful to the reviewer.
Optional, delete this section if not needed.
-->
Includes changes from feat/plugins-api branch.

fvalladaresj committed:
implement front for plugin system

fvalladaresj committed:
Merge branch 'develop' of https://github.com/DashAISoftware/DashAI into feature/front-plugins

fvalladaresj committed:
Merge branch 'feat/plugins-api' of https://github.com/DashAISoftware/DashAI into feature/front-plugins

fvalladaresj committed:
api connection

fvalladaresj committed:
update sampledb to current plugin format

pbadillatorrealba commented:
@fvalladaresj , sorry for the delay, but I've already started reviewing your pr. 

Could you tell me how to execute your changes?
In particular, if you are using sample_db and how can I make it show up in the front end?

Also, tell me if I only have to check only the front code or check everything?

fvalladaresj committed:
Merge branch 'project/plugins' of https://github.com/DashAISoftware/DashAI into feature/front-plugins

fvalladaresj committed:
update get plugins endpoint

fvalladaresj commented:
@pbadillatorrealba I've updated my branch with the current project/plugins branch and fixed a mistake in the `getplugins` endpoint. Also, I'm not using `sample_db` anymore. 

To test, I follow these steps:
-  From DashAI/front, `yarn install` then `yarn start`.
- From DashAI, `pip install .` then `dashai` to run the backend.

Only the front files need reviewing since the back is now up to date with the project/plugins branch.

cristian-tamblay committed:
Pre-commit fix

cristian-tamblay committed:
Fixed some requirements to pass tests on github

pbadillatorrealba commented on DashAI/front/src/types/plugin.ts:
```diff
@@ -0,0 +1,19 @@
+import type { PluginStatus } from "../pages/plugins/constants/pluginStatus";
+
+interface Tag {
```
The naming convention was missing! It should be called ITag.

```suggestion
interface ITag {
```

pbadillatorrealba commented on DashAI/front/src/types/plugin.ts:
```diff
@@ -0,0 +1,19 @@
+  id: number;
+  name: string;
+  author: string;
+  tags: Tag[];
```
Since you already declared the tags as inteface, wouldn't it be a good idea that the tag array also be of type tag?
```suggestion
  tags: Array<ITag>];
```

pbadillatorrealba commented on requirements.txt:
```diff
@@ -16,6 +16,6 @@ typer>=0.9.0
-transformers>=4.23.1
-sacrebleu>=2.3.1
-sentencepiece>=0.1.97
+transformers==4.23.1
+sacrebleu==2.3.1
+sentencepiece==0.1.97
```
What are the consequences of fixing those requirements?
I would prefer that, if it is not necessary to fix them, that we leave them as they were before.

However, it's not critical, as you will eventually outsource the requirements to each plugin. If you don't want to do it, no problem.

fvalladaresj replied:
This changes where made by @cristian-tamblay.

pbadillatorrealba commented on DashAI/front/src/pages/plugins/constants/pluginStatus.ts:
Is it OK to have this file, on which the interface here depends? 
Wouldn't it be better to formalize the plugin states in the same typescript file?

pbadillatorrealba reviewed (changes requested):
Ok, sorry again for the delay.

I left you some minor comments regarding types and requirements.

I ran the front end, but I could see only the plugins administration interface, but none in particular, since I don't have any in the db.

Anyway, and according to what I saw in your code and in the multiple presentations you have made, I think the front end is in good shape to be merged!

Tell me about the comments I left you and then we can approve.

fvalladaresj committed:
merged PluginStatus into plugin type and fixed imports

fvalladaresj commented:
@pbadillatorrealba  I merged PluginStatus enum into the Plugin type file and fixed all the imports. Also fixed Tag to ITag to follow conventions.

inunezn7 committed:
small change to trigger actions

inunezn7 committed:
eslint config updated

inunezn7 committed:
eslint config file in pre-commit updated

inunezn7 committed:
lint error fixed

inunezn7 committed:
yarn.lock deleted

inunezn7 committed:
Merge branch 'project/plugins' into feature/front-plugins

inunezn7 committed:
eslint version updated to 9.2.0

inunezn7 committed:
yarn.lock updated
